Remotion/ObjectBinding/Validation/BusinessObjectValidationResult.cs

```diff
@@ -34,11 +34,14 @@ public static BusinessObjectValidationResult Create ([NotNull] ValidationResult
     {
       ArgumentUtility.CheckNotNull("validationResult", validationResult);
 
-      var validationFailures = validationResult.Errors.Select(e => (CreateBusinessObjectValidationFailure(e), e));
+      var validationFailures = new List<(BusinessObjectValidationFailure, ValidationFailure)>();
+      foreach (var validationFailure in validationResult.Errors)
+        CreateBusinessObjectValidationFailures(validationFailure, f => validationFailures.Add((f, validationFailure)));
+
       return new BusinessObjectValidationResult(validationFailures);
     }
 
-    private static BusinessObjectValidationFailure CreateBusinessObjectValidationFailure (ValidationFailure validationFailure)
+    private static void CreateBusinessObjectValidationFailures (ValidationFailure validationFailure, Action<BusinessObjectValidationFailure> addFailure)
```

Comment:
I would like to discuss the pro/cons of this approach vs the linq-based approach. it still feels clunky to me.

Reply:
decision: Revert to `IEnumerable<T>` approach, but extract creation code into own methods.